src/app/(pages)/fhirServers/page.tsx

```diff
@@ -165,6 +187,35 @@ const FhirServers: React.FC = () => {
     };
   }, []);
 
+  const addHeader = () => {
+    setHeaders([
+      ...headers,
+      { key: "", value: "", id: `${Date.now()}-${Math.random()}` },
+    ]);
+  };
+
+  const updateHeader = (id: string, field: "key" | "value", value: string) => {
+    setHeaders(
+      headers.map((header) =>
+        header.id === id ? { ...header, [field]: value } : header,
+      ),
+    );
+  };
+
+  const removeHeader = (id: string) => {
+    setHeaders(headers.filter((header) => header.id !== id));
+  };
+
+  const convertHeadersToObject = (): Record<string, string> => {
```

Comment:
just for my understanding, what's the advantage of storing the headers as an array and then normalizing it into an object rather than just storing the headers as an object? I think it's because it makes the appending process for additional headers easier (since we don't have to validate the keys in the final dict until we're ready to save) but wanted to check my understanding

Reply:
Correct - it was just easier, especially since I'm using `Object.entries` to parse the headers object stored in the DB, and that returns an array of arrays, it was easier to just map that to an array of HeaderPair objects, which could also store the unique ID for the React key prop, which we could then throw away when converting back to an object. Also deleting from arrays is much easier than deleting keys from objects!